src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -893,8 +1049,38 @@ def forward(
 
         logits = self.lm_head(hidden_states)
 
+        loss = None
+        if labels is not None:
+
+            # retrieve loss input_lengths from attention_mask
+            attention_mask = (
+                attention_mask if attention_mask is not None else torch.ones_like(input_values, dtype=torch.long)
+            )
+            input_lengths = self._get_feat_extract_output_lengths(attention_mask.sum(-1))
+
+            # assuming that padded tokens are filled with -100
+            # when not being attended to
+            labels_mask = labels >= 0
```

Comment:
That's important we assume that padded labels have been replaced by -100 before training or else the fine-tuning would crash. The second option would be to actually check whether the labels contain a `pad_token_id` and to automatically replace this token then. However I think the first option is more in line with how we currently do fine-tuning.

Reply:
I agree the first option. We don't replace -100 with pad for any model, so I think the first option is good.